.circleci/config.yml

```diff
@@ -0,0 +1,128 @@
+jobs:
+  test:
+    macos:
+      xcode: "9.4.0"
+    steps: 
+      - checkout
+      - run:
+          name: tagdescription
+          command: |
+              echo "test step1"
+              echo "test step2"
+
+  release_carthage:
+    # Specify the Xcode version to use
+    macos:
+      xcode: "9.2.0"
+
+    steps: 
+      - checkout
+      - run:
+          name: upgrade carthage
+          command: brew outdated carthage || brew upgrade carthage
+      - run:
+          name: install github-release
+          command: brew install github-release
+      - run:
+          name: Build Cathage
+          command: carthage build --no-skip-current
+
+      - run:
+          name: Create Cathage Archive
+          command: carthage archive  
+
+      - run:
+          name: Rename Cathage Achive
+          command:  mv  AWSAPIGateway.framework.zip   aws-sdk-ios-carthage.framework.zip
+
+      - run:
+          name: release the tag
+          command: |
+            tagdescription=$(sed -n '/## '${CIRCLE_TAG}'/,/## [0-9]*\.[0-9]*\.[0-9]/p'  CHANGELOG.md | sed '1d' | sed '$d')
+            tagname="AWS SDK for iOS "${CIRCLE_TAG}
+            github-release release -s ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME}  -t ${CIRCLE_TAG} -d "$tagdescription" -n "$tagname"         
+      - run:
+          name: upload file to git release 
+          command: github-release upload -s ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME}  -t ${CIRCLE_TAG} -f aws-sdk-ios-carthage.framework.zip -n aws-sdk-ios-carthage.framework.zip
+
+  release_cocoapods:
+    # Specify the Xcode version to use
+    macos:
+      xcode: "9.4.0"
+    steps:
+      - checkout  
+       # Install CocoaPods
+    #  - run:
+     #     name: pod install
+      #    command: pod install
+      
+      - run:
+          name: Release cocoapods
+          command : python3    cocoapods_release.py
+  release_appledoc:
+    # Specify the Xcode version to use
+    macos:
+      xcode: "9.2.0"
+    steps:
+      - checkout  
+       # Install CocoaPods
+      - run:
+          name: Install appledoc 
+          command: brew install appledoc
+          # curl -o install-appledoc.sh https://raw.githubusercontent.com/tomaz/appledoc/master/install-appledoc.sh
+          # bash install-appledoc.sh -t default          
+      - run:
+          name: generate documents
+          command: bash ./Scripts/GenerateAppleDocs.sh
+
+      - run:
+          name: copy documents
+          command: |
+            git checkout -f gh-pages
+            cp -R docs/html/*   docs/reference/
+
+      - run:
+          name: checkin documents
+          command: |
+            git add docs
+            git commit -m "Update document via CircleCI"
+            git push -q https://${GITHUB_TOKEN}@github.com/aws/aws-sdk-ios.git  
+
+      # Build the app and run tests     
+      
+
+
+
+workflows:
+  version: 2
+  build_and_test:
+    jobs: 
+      - test:
```

Comment:
since we are running the `test` workflow on master, we can probably add an enhancement which is to build the whole SDK for every commit merged into the SDK. The cocoapods and carthage are correctly tied to a tag but we can do a standard build with Unit tests on all builds.

Reply:
the test job will be removed